pkg/collector/corechecks/ebpf/probe/ebpfcheck/probe.go

```diff
@@ -523,3 +537,313 @@ func ringBufferMemoryUsage(mapStats *model.EBPFMapStats, info *ebpf.MapInfo, k *
+		ElemFlags uint64
+		Flags     uint64
+	}
+
+	// Allocate the buffers we need, and check if we got enough for getting
+	// all the entries in a single batch or if we reached the limit and we need to
+	// iterate
+	allocatedEnoughSpace := buffers.tryEnsureSizeForFullBatch(mp)
+	if !allocatedEnoughSpace {
+		buffers.prepareFirstBatchKeys(mp)
+	}
+
+	batchSize := min(mp.MaxEntries(), uint32(len(buffers.keys)))
+	totalCount := int64(0)
+
+	// To avoid inifinte loops, we limit the number of restarts and we
+	for restarts, batchIndex := 0, uint32(0); restarts < maxRestarts && batchIndex*batchSize < mp.MaxEntries(); batchIndex++ {
+		// Ensure that we get a different hash if we get restarted, so false positives caused by hash collisions can be mitigated
+		buffers.firstBatchKeys.seed = uint32(restarts)
+
+		// Prepare the arguments to the lookup call
+		attr := MapLookupBatchAttr{
+			MapFd:    uint32(mp.FD()),
+			Values:   NewPointer(unsafe.Pointer(&buffers.values[0])),
+			Keys:     NewPointer(unsafe.Pointer(&buffers.keys[0])),
+			Count:    batchSize,
+			OutBatch: NewPointer(unsafe.Pointer(&buffers.cursor[0])),
+			InBatch:  NewPointer(nil), // nil means start at the beginning
+		}
+		if batchIndex == 0 {
+			// First batch, start at the beginning
+			attr.InBatch = NewPointer(nil)
+		} else {
+			// continue from where we left off
+			attr.InBatch = NewPointer(unsafe.Pointer(&buffers.cursor[0]))
+		}
+
+		_, _, errno := unix.Syscall(unix.SYS_BPF, uintptr(BpfMapLookupBatchCommandCode), uintptr(unsafe.Pointer(&attr)), unsafe.Sizeof(attr))
+		totalCount += int64(attr.Count)
+
+		if errno == 0 && batchIndex == 0 {
+			// We got a batch and it's the first one, and we didn't reach the end of the map, so we need to store the keys we got here
+			// so that later on we can check against them to see if we got an iteration restart
+			if !allocatedEnoughSpace { // A sanity check
+				return -1, fmt.Errorf("Unexpected batch lookup result: we should have enough space to get the full map in one batch, but BatchLookup returned a partial result")
+			}
+
+			// Keep track the keys of the first batch so we can look them up later to see if we got restarted
+			buffers.firstBatchKeys.load(buffers.keys, int(attr.Count))
+		} else if batchIndex > 0 {
+			// We got a batch and it's not the first one. Check against the keys received in the first batch
+			// to see if we got an iteration restart
+			if buffers.firstBatchKeys.containsAny(buffers.keys, int(attr.Count)) {
+				// We got a restart, reset the counters and start from this batch as if were the first
+				buffers.firstBatchKeys.load(buffers.keys, int(attr.Count))
+				restarts++
+				batchIndex = 0
+				totalCount = 0
+				continue
+			}
+		}
+
+		if errno == unix.ENOENT {
+			// We looked up all elements, count is valid, return it
+			return totalCount, nil
+		} else if errno != 0 {
+			// Something happened, abort everything
+			return -1, fmt.Errorf("error iterating map %s: %s", mp.String(), errno)
```

Comment:
The error name can sometimes be more helpful than the descriptive string, so let's add that here.
```suggestion
			return -1, fmt.Errorf("error iterating map %s: %s: %s", mp.String(), unix.ErrnoName(errno), errno)
```